packages/web-lib/hooks/useChain.ts

```diff
@@ -0,0 +1,35 @@
+import {chains, TChain} from '@yearn-finance/web-lib/utils/web3/chains';
+import { TNDict } from '../utils/types';
+import { useChainID } from './useChainID';
+
+type TUseChainReturn = {
+	get: (chainID: number) => TChain;
+	getCurrent: () => TChain;
+	getAll: () => TNDict<TChain>;
+}
+
+/* 🔵 - Yearn Finance ******************************************************
+** This hook can be used to grab details about a network.
+** It will return details of the network.
+**************************************************************************/
+export function useChain(): TUseChainReturn {
+	const {safeChainID} = useChainID();
+
+	return {
+		get: (chainID: number): TChain => {
+			if(!chains[chainID]) {
+				throw new Error(`Chain with id ${chainID} is not available`)
+			}
+
+			return chains[chainID];
+		},
+		getCurrent: (): TChain => {
+			if(!chains[safeChainID]) {
+				throw new Error(`Chain with id ${safeChainID} is not available`)
+			}
```

Comment:
It's a very good point, refactored to return `null` instead of throwing with https://github.com/yearn/web-lib/pull/164/commits/398830be88a2ff4722290e871ae5c1262c36959f